rectools/models/sasrec.py

```diff
@@ -462,10 +674,37 @@ def process_dataset_train(self, dataset: Dataset) -> Dataset:
         user_id_map = IdMap.from_values(interactions[Columns.User].values)
         item_id_map = IdMap.from_values(self.item_extra_tokens)
         item_id_map = item_id_map.add_ids(interactions[Columns.Item])
+
+        # get item features
+        prepared_item_features = None
+        if dataset.item_features is not None:
+            item_features = dataset.item_features
+            if not isinstance(item_features, SparseFeatures):
+                raise ValueError("`item_features` in `dataset` must be `SparseFeatures` instance.")
+
+            item_features_names = item_features.names
+
+            internal_ids = dataset.item_id_map.convert_to_internal(interactions[Columns.Item].unique())
+            item_features_by_sorted_interactions = item_features.take(internal_ids)
+
+            dtype = item_features_by_sorted_interactions.values.dtype
+            num_features = item_features_by_sorted_interactions.values.shape[1]
+            pad_item_features = sp.csr_matrix((1, num_features), dtype=dtype)
+
+            item_features_values: sp.scr_matrix = sp.bmat(
+                [[pad_item_features.toarray()], [item_features_by_sorted_interactions.values]], format="csr"
+            )
+
+            prepared_item_features = SparseFeatures.from_iterables(
+                values=item_features_values, names=item_features_names
+            )
+
         interactions = Interactions.from_raw(interactions, user_id_map, item_id_map)
-        dataset = Dataset(user_id_map, item_id_map, interactions)
+
+        dataset = Dataset(user_id_map, item_id_map, interactions, item_features=prepared_item_features)
 
         self.item_id_map = dataset.item_id_map
+        self.item_features = prepared_item_features
```

Comment:
`transform_dataset_u2i` and `transform_dataset_i2i` are important because they are used afterwards in ModelBase code to create target users and items in internal ids. And user interactions are also passed to recommend dataloader.
But both methods use only id mapping, no actual data of previous interactions or features. Saving id map to the model during fitting is fine. Saving actual data is not good if it can be avoided.